File: pacific-adapter/demo/src/main/java/com/example/demo/MainActivity.java
package com.example.demo;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v7.app.AppCompatActivity;

import com.example.demo.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;
    FragmentAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        adapter = new FragmentAdapter(getSupportFragmentManager());
        binding.viewPager.clearOnPageChangeListeners();
        binding.layoutTab.clearOnTabSelectedListeners();
        binding.layoutTab.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                binding.viewPager.setCurrentItem(tab.getPosition(), false);
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {
            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {
            }
        });
        binding.viewPager.addOnPageChangeListener(
                new TabLayout.TabLayoutOnPageChangeListener(binding.layoutTab));
        binding.layoutTab.setupWithViewPager(binding.viewPager);
        binding.viewPager.setAdapter(adapter);
    }
}

File: pacific-adapter/demo/src/main/java/com/example/adapter/MainActivity.kt
package com.example.adapter

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v7.app.AppCompatActivity
import com.example.adapter.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    internal val binding: ActivityMainBinding by lazy {
        DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
    }
    internal val adapter = FragmentAdapter(supportFragmentManager)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding.viewPager.clearOnPageChangeListeners()
        binding.layoutTab.clearOnTabSelectedListeners()
        binding.layoutTab.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                binding.viewPager.setCurrentItem(tab.position, false)
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {}

            override fun onTabReselected(tab: TabLayout.Tab) {}
        })
        binding.viewPager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(binding.layoutTab))
        binding.layoutTab.setupWithViewPager(binding.viewPager)
        binding.viewPager.adapter = adapter
    }
}
